apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts

```diff
@@ -880,10 +880,18 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
 
                     if (!this.isProformaInvoice && !this.isEstimateInvoice) {
                         if (this.isSalesInvoice || this.isCashInvoice || this.isCreditNote || this.isDebitNote) {
-                            this.store.dispatch(this.invoiceReceiptActions.getVoucherDetailsV4(this.accountUniqueName, {
-                                invoiceNumber: this.invoiceNo,
-                                voucherType: this.proformaInvoiceUtilityService.parseVoucherType(this.invoiceType)
-                            }));
+                            if (this.voucherApiVersion === 2) {
+                                this.store.dispatch(this.invoiceReceiptActions.getVoucherDetailsV4(this.accountUniqueName, {
+                                    invoiceNumber: this.invoiceNo,
+                                    voucherType: this.proformaInvoiceUtilityService.parseVoucherType(this.invoiceType),
+                                    uniqueName: this.selectedItem?.uniqueName
+                                }));
+                            } else {
+                                this.store.dispatch(this.invoiceReceiptActions.getVoucherDetailsV4(this.accountUniqueName, {
+                                    invoiceNumber: this.invoiceNo,
+                                    voucherType: this.proformaInvoiceUtilityService.parseVoucherType(this.invoiceType)
```

Comment:
Consider adding the `uniqueName` with conditional operator
```suggestion
                                    voucherType: this.proformaInvoiceUtilityService.parseVoucherType(this.invoiceType),
                                    uniqueName: this.voucherApiVersion === 2 ? this.selectedItem?.uniqueName : undefined
```